recipes/cpr/all/conanfile.py

```diff
@@ -1,63 +1,76 @@
     url = "https://github.com/conan-io/conan-center-index"
-    description = "Keep it short"
-    license = "https://github.com/whoshuu/cpr/blob/1.3.0/LICENSE"
     homepage = "https://whoshuu.github.io/cpr/"
+    license = "MIT"
+    topics = ("conan", "cpr", "requests", "web", "curl")
     exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False],
-               "use_ssl": [True, False],
-               "fPIC": [True, False]}
-    default_options = {"shared": False,
-                       #"libcurl:with_ldap": False,
-                       "use_ssl": True,
-                       "fPIC": True}
-
-    generators = "cmake", "cmake_find_package"
-    _source_subfolder = "cpr"
-
-    def requirements(self):
-        self.requires("libcurl/7.66.0")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if self.options.use_ssl:
-            self.options["libcurl"].with_openssl = True
-        else:
-            self.options["libcurl"].with_openssl = False
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        os.rename("cpr-{}".format(self.version), self._source_subfolder)
+
+    def requirements(self):
+        self.requires("libcurl/7.67.0")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1d")
 
+    def _patch_sources(self):
         for patch in self.conan_data["patches"][self.version]:
             tools.patch(**patch)
 
-    def build(self):
+    def _configure_cmake(self):
         cmake = CMake(self)
-        cmake.definitions["USE_SYSTEM_CURL"] = True # Force CPR to not try to build curl itself from a git submodule
+        cmake.definitions["USE_SYSTEM_CURL"] = True
         cmake.definitions["BUILD_CPR_TESTS"] = False
         cmake.definitions["GENERATE_COVERAGE"] = False
         cmake.definitions["USE_SYSTEM_GTEST"] = False
-        cmake.configure()
+        cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.with_openssl
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        if self.options.with_openssl and not self.options["libcurl"].with_openssl:
+            raise ConanInvalidConfiguration("libcurl must be built with openssl support")
+
+        self._patch_sources()
```

Comment:
Why applying patches in the `build()` method instead of `source` if these patches should be applied for all the configurations?

Reply:
True, this is a convention that most recipes try to follow.
By doining it in the `build()` method, all patching can happen in one function (also setting and option specific)

Reply:
Ok, understood. I'm not sure if there is a policy about it, but, if needed, we can agree on it in the PR to conan-center-index